tests/php/ORM/DBSchemaManagerTest.php

```diff
@@ -48,30 +49,18 @@ public function provideCanCheckAndRepairTable()
         ];
     }
 
-    /**
-     * @dataProvider provideCanCheckAndRepairTable
-     */
+    #[DataProvider('provideCanCheckAndRepairTable')]
     public function testCanCheckAndRepairTable(string $tableName, bool $checkAndRepairOnBuild, bool $expected)
     {
         // set config
         Config::modify()->set(DBSchemaManager::class, 'check_and_repair_on_build', $checkAndRepairOnBuild);
         Config::modify()->set(DBSchemaManager::class, 'exclude_models_from_db_checks', [ChildClass::class]);
 
-        $manager = $this->getConcreteSchemaManager();
+        $manager = $this->getMockBuilder(DBSchemaManager::class)->getMock();
         $reflectionCanCheck = new ReflectionMethod($manager, 'canCheckAndRepairTable');
         $reflectionCanCheck->setAccessible(true);
         $result = $reflectionCanCheck->invoke($manager, $tableName);
 
         $this->assertSame($expected, $result);
     }
-
-    /**
-     * DBSchemaManager is an abstract class - this gives us an instance of a concrete subclass.
-     * This allows us to test the original abstract implementations of methods without risking accidentally
-     * testing overridden methods on something like MySQLSchemaManager
-     */
-    private function getConcreteSchemaManager(): DBSchemaManager
-    {
-        return $this->getMockBuilder(DBSchemaManager::class)->getMockForAbstractClass();
```

Comment:
getMockForAbstractClass() is deprecated https://github.com/sebastianbergmann/phpunit/issues/5241

I'm not sure why this test is worried about things being overridden by child classes as we're getting a mock of the base class using phpunit, not Silverstripe Injector

I've simply switch this to use ->getMock() instead